Bug: math domain error in _orbital_rotation_jw when decomposing OrbitalRotationJW

### Issue
When decomposing an `OrbitalRotationJW` gate, ffsim calls `_orbital_rotation_jw(...)`, which in turn calls `XXPlusYYGate(2 * math.acos(c), cmath.phase(s) - 0.5 * math.pi)` for each Givens rotation. The Givens decomposition can return a cosine `c` that is sligthly outside [-1,1] (like 1.000000000001 for instance) due to floating-point roundoff, causing `math.acos(c)` to raise math domain error.

### Suggested fix

See my latest PR https://github.com/qiskit-community/ffsim/pull/501

> Do you have an example that triggers it? See [#501 (comment)](https://github.com/qiskit-community/ffsim/pull/501#issuecomment-3512433174).

Yeah, I was just simulating the LUCJ ansatz, and after appending HF state and UCJOpSpinBalancedJW to a circuit and running circ.decompose() for some repetitions, it hit the error. Take a look at the test function in the PR I added now. It fails without the clamping.

```
    def _orbital_rotation_jw(
        qubits: Sequence[Qubit], orbital_rotation: np.ndarray
    ) -> Iterator[CircuitInstruction]:
        givens_rotations, phase_shifts = linalg.givens_decomposition(orbital_rotation)
        for c, s, i, j in givens_rotations:
            yield CircuitInstruction(
>               XXPlusYYGate(2 * math.acos(c), cmath.phase(s) - 0.5 * math.pi),
                                 ^^^^^^^^^^^^
                (qubits[i], qubits[j]),
            )
E           ValueError: math domain error

.venv/lib/python3.12/site-packages/ffsim/qiskit/gates/orbital_rotation.py:200: ValueError
```

--- a/python/ffsim/linalg/givens.py
+++ b/python/ffsim/linalg/givens.py
@@ -202,8 +202,10 @@ def givens_decomposition(
     # convert left rotations to right rotations
     for c, s, i, j in reversed(left_rotations):
         c, s = zrotg(c * current_matrix[j, j], s.conjugate() * current_matrix[i, i])
-        # clamp c between -1 and 1 to account for floating point error
-        right_rotations.append(GivensRotation(min(1, max(-1, c)), -s.conjugate(), i, j))
+        right_rotations.append(
+            # clamp c between -1 and 1 to account for floating point error
+            GivensRotation(min(1.0, max(-1.0, c)), -s.conjugate(), i, j)
+        )
 
         givens_mat = np.array([[c, -s], [s.conjugate(), c]])
         givens_mat[:, 0] *= current_matrix[i, i]

--- a/python/ffsim/qiskit/gates/slater_determinant.py
+++ b/python/ffsim/qiskit/gates/slater_determinant.py
@@ -348,7 +348,7 @@ def _givens_decomposition_slater(orbital_coeffs: np.ndarray) -> list[GivensRotat
                 # zero out element j of row i
                 c, s = zrotg(current_matrix[i, j - 1], current_matrix[i, j])
                 # clamp c between -1 and 1 to account for floating point error
-                rotations.append(GivensRotation(min(1, max(-1, c)), s, j, j - 1))
+                rotations.append(GivensRotation(min(1.0, max(-1.0, c)), s, j, j - 1))
                 (
                     current_matrix[:, j - 1],
                     current_matrix[:, j],

